main/src/test/java/org/mobilitydata/gtfsvalidator/validator/BlockTripsWithOverlappingStopTimesValidatorTest.java

```diff
@@ -0,0 +1,198 @@
+                .setStopSequence(stopSequence)
+                .setStopId(stopId)
+                .build();
+    }
+
+    public static GtfsTrip createTrip(long csvRowNumber, String tripId,
+                                      String serviceId, String blockId) {
+        return new GtfsTrip.Builder()
+                .setCsvRowNumber(csvRowNumber)
+                .setTripId(tripId)
+                .setServiceId(serviceId)
+                .setBlockId(blockId)
+                .build();
+    }
+
+    private static GtfsTripTableContainer
+    createTripTable(String[] tripIds, String[] serviceIds, String blockId,
+                    NoticeContainer noticeContainer) {
+        Preconditions.checkArgument(
+                tripIds.length == serviceIds.length,
+                "tripIds.length must be equal to serviceIds.length");
+
+        ArrayList<GtfsTrip> trips = new ArrayList<>();
+        trips.ensureCapacity(tripIds.length);
+        for (int i = 0; i < tripIds.length; ++i) {
+            trips.add(
+                    createTrip(trips.size() + 1, tripIds[i], serviceIds[i], blockId));
+        }
+
+        return GtfsTripTableContainer.forEntities(trips, noticeContainer);
+    }
+
+    private static GtfsStopTimeTableContainer
+    createStopTimeTable(String[] tripIds, String[] stopIds, String[][] times,
+                        NoticeContainer noticeContainer) {
+        Preconditions.checkArgument(tripIds.length == times.length,
+                "tripIds.length must be equal to times.length");
+
+        ArrayList<GtfsStopTime> stopTimes = new ArrayList<>();
+        stopTimes.ensureCapacity(tripIds.length * stopIds.length);
+        for (int i = 0; i < tripIds.length; ++i) {
+            Preconditions.checkArgument(
+                    stopIds.length == times[i].length,
+                    "stopIds.length must be equal to times[%d].length", i);
+            for (int j = 0; j < stopIds.length; ++j) {
+                stopTimes.add(createStopTime(stopTimes.size() + 1, tripIds[i],
+                        times[i][j], stopIds[j], j));
+            }
+        }
+
+        return GtfsStopTimeTableContainer.forEntities(stopTimes, noticeContainer);
+    }
+
+    @Test
+    public void goodFeed() {
+        final NoticeContainer noticeContainer = new NoticeContainer();
+
+        BlockTripsWithOverlappingStopTimesValidator validator =
+                new BlockTripsWithOverlappingStopTimesValidator();
+        validator.calendarTable = createCalendarTable(noticeContainer);
+        validator.calendarDateTable =
+                GtfsCalendarDateTableContainer.forMissingFile();
+        validator.tripTable = createTripTable(
+                new String[]{"t0", "t1", "t2", "t3", "t4", "t5"},
+                new String[]{"WEEK", "WEEK", "WEEK", "SAT", "SAT", "SAT"}, "b1",
+                noticeContainer);
+        validator.stopTimeTable =
+                createStopTimeTable(new String[]{"t0", "t1", "t2", "t3", "t4", "t5"},
```

Comment:
I think we are covered in the case where there is a single entry in stop_times.txt for a trip, IIRC? If not please address in a separate PR.

Reply:
This is a very good point. Let's cover those cases now: #598

We can add a validator for amount of stops later.

Reply:
> We can add a validator for amount of stops later.

Just opened an issue on this matter, see https://github.com/MobilityData/gtfs-validator/issues/599. Thanks @aababilov.